### DISCUSSION TITLE
Writing a module to the Cellar site-packages with cmake

### DISCUSSION THREAD

**Discussion by danielverdugo** [Homebrew/discussions#433] [Everyday usage]
I'm making a C++ formula which builds from source with `cmake` and  tries to install a .so library to `site-packages` directory of my Homebrew Python3.9 instance in Cellar, but fails with `Operation not permitted.` error.

I guess, it's something to do with the sandbox protocol.

I wonder how do I make it work?

Building fails because of the error.

Couldn't find my answer at https://docs.brew.sh/Python-for-Formula-Authors

**Comment by gromgit** [comment]
> I wonder how do I make it work?

By installing your library into the `libexec` directory of _your own formula_, then create a `pth` file to add the `libexec` directory to Python's `sys.path` at runtime. See the `csound` formula as an example.

> Couldn't find my answer at https://docs.brew.sh/Python-for-Formula-Authors

https://docs.brew.sh/Python-for-Formula-Authors#installing-1
